Sources/Mixpanel.swift

```diff
@@ -56,6 +56,49 @@ open class Mixpanel {
                                                          superProperties: superProperties,
                                                          serverURL: serverURL)
     }
+    
+    /**
+     Initializes an instance of the API with the given project token.
+     
+     Returns a new Mixpanel instance API object. This allows you to create more than one instance
+     of the API object, which is convenient if you'd like to send data to more than
+     one Mixpanel project from a single app.
+     
+     - parameter token:                     your project token
+     - parameter trackAutomaticEvents:      Whether or not to collect common mobile events
+     - parameter flushInterval:             Optional. Interval to run background flushing
+     - parameter instanceName:              Optional. The name you want to uniquely identify the Mixpanel Instance.
+     It is useful when you want more than one Mixpanel instance under the same project token.
+     - parameter optOutTrackingByDefault:   Optional. Whether or not to be opted out from tracking by default
+     - parameter useUniqueDistinctId:       Optional. whether or not to use the unique device identifier as the distinct_id
+     - parameter superProperties:           Optional. Super properties dictionary to register during initialization
+     - parameter proxyServerConfig:         Optional. Setup for proxy server.
+     
+     - important: If you have more than one Mixpanel instance, it is beneficial to initialize
+     the instances with an instanceName. Then they can be reached by calling getInstance with name.
+     
+     - returns: returns a mixpanel instance if needed to keep throughout the project.
+     You can always get the instance by calling getInstance(name)
+     */
+    
+    @discardableResult
+    open class func initialize(token apiToken: String,
+                               trackAutomaticEvents: Bool,
+                               flushInterval: Double = 60,
+                               instanceName: String? = nil,
+                               optOutTrackingByDefault: Bool = false,
+                               useUniqueDistinctId: Bool = false,
+                               superProperties: Properties? = nil,
+                               proxyServerConfig: ProxyServerConfig) -> MixpanelInstance {
```

Comment:
if `proxyServerConfig` is optional, could you add `proxyServerConfig: ProxyServerConfig? = nil` to the existing `initialize` function rather than adding a new function?

Reply:
Thanks for the feedback on the PR 🙌 .  On using the same `intialize` function, they are few points I want to highlight.

- By combining `serverURL` and `ProxyServerConfig` into one method, we risk users setting both parameters, which doesn't align with their intended use. Ideally one of them should be set. Keeping them separate ensures users have a clear choice without the possibility of configuring conflicting options.
- With two distinct initialization methods, we're directly guiding users on their options. **One method is for basic server URL configuration, and the other is for more advanced proxy configurations.**